packages/navuser/src/react/index.js

```diff
@@ -0,0 +1,72 @@
+import Avatar from '@pluralsight/ps-design-system-avatar'
+import filterReactProps from '@pluralsight/ps-design-system-filter-react-props'
+import Halo from '@pluralsight/ps-design-system-halo'
+import { css } from 'glamor'
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import stylesheet from '../css/index.js'
+
+const styles = {
+  navUser: props =>
+    css(
+      stylesheet['.psds-navuser'],
+      (props.href || props.onClick) && stylesheet['.psds-navuser--clickable']
+    ),
+  planName: () => css(stylesheet['.psds-navuser__plan-name']),
+  name: () => css(stylesheet['.psds-navuser__name']),
+  words: () => css(stylesheet['.psds-navuser__words'])
+}
+
+const NavUser = React.forwardRef((props, forwardedRef) => {
+  const tagName = props.href ? 'a' : props.onClick ? 'button' : 'div'
+
+  const ref = React.useRef()
+  React.useImperativeHandle(forwardedRef, () => ref.current)
+
+  return (
+    <Halo inline gapSize={Halo.gapSizes.small}>
+      {React.createElement(
+        tagName,
+        {
+          ...filterReactProps(props, { tagName }),
+          ...styles.navUser(props),
+          ref
+        },
+        <>
+          <Avatar
+            src={props.src}
+            size={Avatar.sizes.xSmall}
+            name={props.name}
+          />
+          <Words {...props} />
+        </>
+      )}
+    </Halo>
+  )
+})
+NavUser.displayName = 'NavUser'
+NavUser.propTypes = {
+  href: PropTypes.string,
+  onClick: PropTypes.func,
+  planName: PropTypes.string,
```

Comment:
is this always going to be the plan name. is it more like a secondary text or subtitle area

Reply:
I made a call here.  Still not sure it's the right one.  

I called it planName because:

- Most foundationally, I started thinking about the NavUser component as a "temporary" DS component.  It'll help product teams get over the transition to product navs.  Then it'll be owned by someone else, likely the profile team, at the point that team has an opinion on how it should evolve.  At that point, if they believe that the plan name should not appear there or the space should be repurposed, they can. It'll probably be a breaking change to the api, and life will go on. We won't own it beyond the time it's `planName`.
- User `name` is also a semantic prop that I think is nice.  Easier to know what goes there.  
- Suggesting semantics through the API will help with consistent use.

Reply:
cc @sweeting ... I see that you updated the figma to read "primary" and "secondary" again this morning.  Will you please weigh in on your current thinking and respond to my reasoning above?

Reply:
@jaketrent Even if it is temporary, I like the primary and secondary naming for those lines of text.

Reply:
@sweeting Why do you like "primary" and "secondary" for naming those properties?

Reply:
^ bump.  @sweeting I'm still interested in continuing this ^ discussion. I'm interested in your reasoning on the naming preference. I believe this is the last point of discussion around the current PR before merge.

Reply:
@jaketrent I was just thinking it might be better to be less specific. `name` is probably fine since that's likely all it would be used for. `planName` may be more fraught since sometimes it's referred to internally and externally both as a "plan" and "account". Maybe we just rename `planName` to `meta`. That way it could be the the name of a plan or account or even used for an email address if that's what they decide to do.